### PR TITLE
Bug 1942164: Increase OVN upgrade timeout to 90m from 75m

### PR DESCRIPTION
**Pull request by jluhrsen**
The OVN upgrade jobs almost always take longer than 75m
so this fails. There is more context to this here:
  https://bugzilla.redhat.com/show_bug.cgi?id=1942164

Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

**Comment by jluhrsen** [comment]
/retitle Bug 1942164: Increase upgrade timeout to 90m from 75m

**Comment by vrutkovs** [comment]
https://github.com/openshift/origin/pull/26219 would extend soft limit on upgrade time. Lets rework this PR to do the same for OVN (probably extending OVN on AWS configuration even more)

**Comment by stbenjam** [comment]
I think we want this to only bump it for OVN, like we did only for AWS: https://github.com/openshift/origin/pull/26219

**Comment by jluhrsen** [comment]
@vrutkovs , @stbenjam , please see the latest version of this effort. basically, if it's AWS it should end up at 105m and if it's not AWS then 75 for OpenShiftSDN and 90 for OVN.

File: test/e2e/upgrade/upgrade.go [hunk 1]
```diff
@@ -270,14 +270,28 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
...
 	framework.ExpectNoError(err)
 	if infra.Status.PlatformStatus.Type == configv1.AWSPlatformType {
 		// due to https://bugzilla.redhat.com/show_bug.cgi?id=1943804 upgrades take ~12 extra minutes on AWS
-		durationToSoftFailure = 105 * time.Minute
+		// and see commit d69db34a816f3ce8a9ab567621d145c5cd2d257f which notes that some AWS upgrades can
+		// take close to 105 minutes total (75 is base duration, so adding 30 more if it's AWS)
+		durationToSoftFailure = (baseDurationToSoftFailure + 30) * time.Minute
+	} else {
+		// if the cluster is on AWS we've already bumped the timeout enough, but if not we need to check if
+		// the CNI is OVN and increase our timeout for that
```
Comment (wking) [inline test/e2e/upgrade/upgrade.go:285]: OVN and provider-LB delay sources seem orthogonal.  If that's right, can we use:

```go
if infra.Status.PlatformStatus.Type == configv1.AWSPlatformType {
  // due to https://bugzilla.redhat.com/show_bug.cgi?id=1943804 upgrades take ~12 extra minutes on AWS
  durationToSoftFailure += 20 * time.Minute
}
network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
framework.ExpectNoError(err)
if network.Status.NetworkType == "OVNKubernetes" {
  durationToSoftFailure += 15 * time.Minute
}
```

or similar?

Comment (jluhrsen) [reply test/e2e/upgrade/upgrade.go:285]: but if it's OVN + AWS we will end up with only +15 when we want +30. I can do the platform check second so that even if it's OVN it will still end up with +30, but that was sort of why I did it this way to start with so I could comment the code about
the AWS increase being good enough for OVN anyway. Let me know what's best though. I'd like to get this in so we can start making small progress to improving that ovn-upgrade job

**Comment by stbenjam** [comment]
OVN upgrades are sitting at 0% mostly because of this, changes look reasonable to me

/assign @bparees @smarterclayton 
/lgtm

**Comment by jluhrsen** [comment]
the upgrade job will never pass

**Comment by jluhrsen** [comment]
this didn't work like I thought. this new PR should clean it up:
  https://github.com/openshift/origin/pull/26324
